Bottom: 1ada6eeadb0d506e78b1a8ad7d0f08443b4fe6d2
Top:    91f421029f80c32d42a9276f218f3b7f38615b76
Author: Jason Wang <jasowang@redhat.com>
Date:   2018-03-28 16:26:23 +0800

vhost-net: batch submit packets to underlayer socket

Signed-off-by: Jason Wang <jasowang@redhat.com>


---

diff --git a/include/linux/if_tun.h b/include/linux/if_tun.h
index c5b0a75..be33526 100644
--- a/include/linux/if_tun.h
+++ b/include/linux/if_tun.h
@@ -19,6 +19,11 @@
 
 #define TUN_XDP_FLAG 0x1UL
 
+struct tun_ubuf_info {
+	int npackets;
+	struct ubuf_info *ubufs;
+};
+
 #if defined(CONFIG_TUN) || defined(CONFIG_TUN_MODULE)
 struct socket *tun_get_socket(struct file *);
 struct ptr_ring *tun_get_tx_ring(struct file *file);
